addons/payment_custom/tests/test_payment_transaction.py

```diff
@@ -0,0 +1,60 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import Command, fields
+from odoo.tests import tagged
+
+from odoo.addons.payment.tests.common import PaymentCommon
+
+
+@tagged('-at_install', 'post_install')
+class TestPaymentTransaction(PaymentCommon):
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+
+        cls.provider = cls._prepare_provider(code='custom')
+        cls.product = cls.env['product.product'].create({
+            'name': "test product", 'list_price': cls.amount
+        })
+
+    def test_custom_communication_for_invoice(self):
```

Comment:
This is clear enough for me since I hardly see what else it could test, but in case you were looking for a more descriptive name:
```suggestion
    def test_communication_based_on_invoice_reference(self):
```

Reply:
I don't think it's clearer. The communication isn't necessarily based on the invoice reference (which I understood as "based on the invoice name") but on the payment reference that's set on the invoice, that might be on the customer of the invoice. Or on whatever is needed by a localization from different information on the invoice. `test_communication_based_on_invoice_payment_reference`

I agree my name is pretty vague though. But I'll let it as is.

Reply:
I won't fight over this but I still want to point out that having 7 words long test names is fine since they're not referenced anywhere, and at least it makes it possible to guess what is broken when browsing the list of failing tests.